release/cc/create_artifacts.py

```diff
@@ -0,0 +1,184 @@
+
+
+def create_targets(container_configs_list, bazel_version):
+  """Create the new docker_toolchain_autoconfig target if not exists.
+
+  An example target located in configs/ubuntu16_04_clang/BUILD is:
+  //configs/ubuntu16_04_clang:msan-ubuntu16_04-clang-1.0-bazel_0.15.0-autoconfig
+
+  There is one target per container per Bazel version per config type.
+
+  The script only creates new targets in the BUILD file if they do not exist.
+
+  Args:
+    container_configs_list: list of ContainerConfigs, the list of
+      ContainerConfigs to generate configs for.
+    bazel_version: string, the version of Bazel used to generate the configs.
+
+  Returns:
+    None
+  """
+
+  container_sha_map = imp.load_source("toolchain_containers", SHA_MAP_FILE)
+
+  for container_configs in container_configs_list:
+
+    # Get the sha256 value of the container used to generate the configs.
+    sha = container_sha_map.toolchain_container_sha256s()[
+        "%s_clang" % container_configs.distro]
+
+    for config in container_configs.configs:
+
+      # Get target basename from config definitions.
+      target = get_autoconfig_target_name(
+          config_type=config.config_type,
+          distro=container_configs.distro,
+          config_version=container_configs.version,
+          bazel_version=bazel_version)
+
+      with open(container_configs.get_target_build_path(), "a+") as build_file:
+        if target not in build_file.read():
+          tpl_file_path = os.path.join(GIT_ROOT, "release", "cc",
+                                       "%s.tpl" % config.config_type)
+          with open(tpl_file_path, "r") as tpl_file:
+            tpl = Template(tpl_file.read()).substitute(
+                DATE=get_date(),
+                DISTRO=container_configs.distro,
+                CONFIG_VERSION=container_configs.version,
+                BAZEL_VERSION=bazel_version,
+                NAME=container_configs.image,
+                SHA=sha)
+
+            build_file.write(tpl)
+
+
+def generate_toolchain_definition(container_configs_list, bazel_version):
+  """Generate new cpp toolchain definitions.
+
+  Example cpp toolchain definitions for clang-ubuntu container are located in
+  configs/ubuntu16_04_clang/1.0/bazel_0.15.0/cpp/.
+
+  There is one BUILD file to contain all cpp toolchain definitions for each
+  config type (e.g. default, msan) per container per Bazel version.
+
+  If the file already exist in this repo, the script will delete it and
+  generate new one.
+
+  Args:
+    container_configs_list: list of ContainerConfigs, the list of
+      ContainerConfigs to generate configs for.
+    bazel_version: string, the version of Bazel used to generate the configs.
+
+  Returns:
+    None
+  """
+
+  for container_configs in container_configs_list:
+
+    cpp_dir = os.path.dirname(container_configs.get_cpp_build_path())
+
+    # Remove old cpp directory if exists.
+    if os.path.isdir(cpp_dir):
```

Comment:
why are you removing the full directory and not just the one file?

Reply:
should you have a warning if deleted as you do for the config generation below

Reply:
This directory only contains one file, and this directory needs to in a new release. So I just remove it if exists, and recreate.

Added the warning.